Mathlib/Data/Matroid/Circuit.lean

```diff
@@ -319,4 +336,128 @@ lemma ext_iff_circuit {M₁ M₂ : Matroid α} :
+  suffices hcl : z ∈ M.closure ((⋃ i, I i) ∪ (J \ {z})) by
+    rw [mem_closure_iff_exists_circuit_of_not_mem (by simp [hzI])] at hcl
+    obtain ⟨C', hC'ss, hC', hzC'⟩ := hcl
+    refine ⟨C', ?_, hC', hzC'⟩
+    rwa [union_comm, ← insert_union, insert_diff_singleton, insert_eq_of_mem hzJ] at hC'ss
+  replace hC := show ∀ (i : ι), M.closure (I i) = M.closure ({x i} ∪ (I i))
+    by simpa [diff_singleton_eq_self (hxI _)] using fun i ↦ (hC i).closure_diff_singleton_eq (x i)
+  -- This is true because each `I i` spans `x i` and `(range x) ∪ (J \ {z})` spans `z`.
+  rw [closure_union_congr_left <| closure_iUnion_congr _ _ hC, iUnion_union_distrib,
+    iUnion_singleton_eq_range, union_right_comm]
+  refine mem_of_mem_of_subset (hJx.mem_closure_diff_singleton_of_mem (.inl hzJ))
+    (M.closure_subset_closure (subset_trans ?_ subset_union_left))
+  rw [union_diff_distrib, union_comm]
+  exact union_subset_union_left _ diff_subset
+
+/-- A generalization of the strong circuit elimination axiom `Matroid.Circuit.strong_elimination`
+to an infinite collection of circuits.
+This version is one of the axioms when defining infinite matroids via circuits.
+
+TODO : A similar statement will hold even when all mentions of `z` are removed. -/
+lemma Circuit.strong_multi_elimination {C₀ : Set α} (C : ι → Set α) (hC₀ : M.Circuit C₀)
+    (hC : ∀ i, M.Circuit (C i))(h_mem_C₀ : ∀ i, x i ∈ C₀) (h_mem : ∀ i, x i ∈ C i)
+    (h_unique : ∀ ⦃i i'⦄, x i ∈ C i' → i = i') (hzC₀ : z ∈ C₀) (hzC : ∀ i, z ∉ C i) :
+    ∃ C' ⊆ (C₀ ∪ ⋃ i, C i) \ range x, M.Circuit C' ∧ z ∈ C' := by
+  have hwin := M.strong_multi_elimination_insert x (fun i ↦ (C i \ {x i}))
+    (J := C₀ \ range x) (z := z) (by simp) (fun i ↦ ?_) ?_ ⟨hzC₀, ?_⟩ ?_
+  · obtain ⟨C', hC'ss, hC', hzC'⟩ := hwin
+    refine ⟨C', hC'ss.trans ?_, hC', hzC'⟩
+    refine union_subset (diff_subset_diff_left subset_union_left)
+      (iUnion_subset fun i ↦ subset_diff.2
+        ⟨diff_subset.trans (subset_union_of_subset_right (subset_iUnion ..) _), ?_⟩)
+    rw [disjoint_iff_forall_ne]
+    rintro _ he _ ⟨j, hj, rfl⟩ rfl
+    obtain rfl : j = i := h_unique he.1
+    simp at he
+  · simpa [insert_eq_of_mem (h_mem i)] using hC i
+  · rwa [diff_union_self, union_eq_self_of_subset_right]
+    rintro _ ⟨i, hi, rfl⟩
+    exact h_mem_C₀ i
+  · rintro ⟨i, hi, rfl⟩
+    exact hzC _ (h_mem i)
+  simp only [mem_diff, mem_singleton_iff, not_and, not_not]
+  exact fun i hzi ↦ (hzC i hzi).elim
+
+/-- The strong circuit elimination axiom. For any two circuits `C₁, C₂` and all `e ∈ C₁ ∩ C₂` and
+`f ∈ C₁ \ C₂`, there is a circuit `C` with `f ∈ C ⊆ (C₁ ∪ C₂) \ {e}`. -/
+lemma Circuit.strong_elimination (hC₁ : M.Circuit C₁) (hC₂ : M.Circuit C₂) (heC₁ : e ∈ C₁)
+    (heC₂ : e ∈ C₂) (hfC₁ : f ∈ C₁) (hfC₂ : f ∉ C₂) :
+    ∃ C ⊆ (C₁ ∪ C₂) \ {e}, M.Circuit C ∧ f ∈ C := by
+  obtain ⟨C, hCs, hC, hfC⟩ := hC₁.strong_multi_elimination (fun i : Unit ↦ e) (fun _ ↦ C₂)
+    (by simpa) (z := f) (by simpa) (by simpa) (by simp) (by simpa) (by simpa)
```

Comment:
The `(z := f)` here suggests that maybe it should be explicit in `strong_multi_elimination`?